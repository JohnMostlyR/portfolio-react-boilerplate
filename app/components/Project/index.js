import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import messages from './messages';
import Article from './Article';
import ProjectBody from './ProjectBody';
import ProjectHeader from './ProjectHeader';

import MediaAsset from '../MediaAsset';
import H3 from '../H3';
import P from '../P';
import Link from '../Link';

class Project extends React.PureComponent {
  componentDidMount() {
    window.scrollTo(0, 0);
  }

  render() {
    const { detailsBodyText, isOdd, search, thumbnail, title } = this.props;

    return (
      <Article>
        <MediaAsset imageSource={thumbnail} isOdd={isOdd} />
        <ProjectBody>
          <div>
            <ProjectHeader>
              <H3>{title}</H3>
            </ProjectHeader>
          </div>
          <P>{detailsBodyText}</P>
          <P textAlign="right">
            <FormattedMessage {...messages.detailsLink}>
              {message => (
                <Link // eslint-disable-line jsx-a11y/anchor-is-valid
                  to={{
                    pathname: `/projects/${title
                      .toLowerCase()
                      .replace(/\W+/g, '-')}/`,
                    search,
                  }}
                  odd={`${isOdd}`}
                >
                  {message}
                </Link>
              )}
            </FormattedMessage>
          </P>
        </ProjectBody>
      </Article>
    );
  }
}

Project.propTypes = {
  detailsBodyText: PropTypes.string,
  isOdd: PropTypes.bool,
  search: PropTypes.string,
  thumbnail: PropTypes.shape({
    file: PropTypes.shape({
      details: PropTypes.shape({
        image: PropTypes.shape({
          height: PropTypes.number,
          width: PropTypes.number,
        }),
      }),
      fileName: PropTypes.string,
      url: PropTypes.string,
    }),
    title: PropTypes.string,
  }).isRequired,
  title: PropTypes.string.isRequired,
};

export default Project;
