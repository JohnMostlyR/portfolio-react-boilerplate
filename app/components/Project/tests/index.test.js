import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Project from '../index';

describe('Project', () => {
  const thumbnail = {
    file: {
      contentType: 'image/png',
      details: {
        image: {
          height: 230,
          width: 345,
        },
        size: 22495,
      },
      fileName: 'medium-m.png',
      url: '/medium-m.png',
    },
    title:
      'Presentation showing how this project looks on mobile, tablet and desktop screens',
  };
  const title = 'This is the title field for Project 1';
  const description = 'This is the description field Project 1';

  it('should render and match the snapshot', () => {
    const wrapper = shallow(
      <Project
        thumbnail={thumbnail}
        title={title}
        detailsBodyText={description}
        isOdd
      />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should adopt the "isOdd" property', () => {
    const wrapper = shallow(
      <Project
        thumbnail={thumbnail}
        title={title}
        detailsBodyText={description}
        isOdd={false}
      />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
