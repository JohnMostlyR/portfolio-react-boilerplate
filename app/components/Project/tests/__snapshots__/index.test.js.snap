// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project should adopt the "isOdd" property 1`] = `
<Article>
  <MediaAsset
    imageSource={
      Object {
        "file": Object {
          "contentType": "image/png",
          "details": Object {
            "image": Object {
              "height": 230,
              "width": 345,
            },
            "size": 22495,
          },
          "fileName": "medium-m.png",
          "url": "/medium-m.png",
        },
        "title": "Presentation showing how this project looks on mobile, tablet and desktop screens",
      }
    }
    isOdd={false}
  />
  <ProjectBody>
    <div>
      <ProjectHeader>
        <H3>
          This is the title field for Project 1
        </H3>
      </ProjectHeader>
    </div>
    <P
      textAlign="left"
    >
      This is the description field Project 1
    </P>
    <P
      textAlign="right"
    >
      <FormattedMessage
        defaultMessage="More details →"
        id="portfolio.components.Project.detailsLink"
        values={Object {}}
      />
    </P>
  </ProjectBody>
</Article>
`;

exports[`Project should render and match the snapshot 1`] = `
<Article>
  <MediaAsset
    imageSource={
      Object {
        "file": Object {
          "contentType": "image/png",
          "details": Object {
            "image": Object {
              "height": 230,
              "width": 345,
            },
            "size": 22495,
          },
          "fileName": "medium-m.png",
          "url": "/medium-m.png",
        },
        "title": "Presentation showing how this project looks on mobile, tablet and desktop screens",
      }
    }
    isOdd={true}
  />
  <ProjectBody>
    <div>
      <ProjectHeader>
        <H3>
          This is the title field for Project 1
        </H3>
      </ProjectHeader>
    </div>
    <P
      textAlign="left"
    >
      This is the description field Project 1
    </P>
    <P
      textAlign="right"
    >
      <FormattedMessage
        defaultMessage="More details →"
        id="portfolio.components.Project.detailsLink"
        values={Object {}}
      />
    </P>
  </ProjectBody>
</Article>
`;
